CSharp/AccountManagement/Program.cs

```diff
@@ -1,334 +1,349 @@
+                Console.WriteLine("  Primary key of account {0}:   {1}", account.Name, accountKeys.Primary);
+                Console.WriteLine("  Secondary key of account {0}: {1}", account.Name, accountKeys.Secondary);
+                Console.WriteLine();
+
+                // Regenerate primary account key
+                Console.WriteLine("Regenerating the primary key of account {0}...", account.Name);
+                BatchAccountRegenerateKeyResult newKeys = await batchManagementClient.Account.RegenerateKeyAsync(
+                    ResourceGroupName, account.Name,
+                    new BatchAccountRegenerateKeyParameters() { KeyName = AccountKeyType.Primary });
+                Console.WriteLine("  New primary key of account {0}: {1}", account.Name, newKeys.Primary);
+                Console.WriteLine("  Secondary key of account {0}:   {1}", account.Name, newKeys.Secondary);
+                Console.WriteLine();
+
+                // Print subscription quota information
+                IPage<AccountResource> listResponse = await batchManagementClient.Account.ListAsync();
+                IList<AccountResource> accounts = listResponse.ToList();
+                Console.WriteLine("Total number of Batch accounts under subscription id {0}:  {1}", batchManagementClient.SubscriptionId, accounts.Count);
+
+                // Determine how many additional accounts can be created in the target region
+                int numAccountsInRegion = accounts.Count(o => o.Location == account.Location);
+                Console.WriteLine("Accounts in {0}: {1}", account.Location, numAccountsInRegion);
+                Console.WriteLine("You can create {0} more accounts in the {1} region.", quotaResponse.AccountQuota - numAccountsInRegion, account.Location);
+                Console.WriteLine();
+
+                // List accounts in the subscription
+                Console.WriteLine("Listing all Batch accounts under subscription id {0}...", batchManagementClient.SubscriptionId);
+                foreach (AccountResource acct in accounts)
+                {
+                    Console.WriteLine("  {0} - {1} | Location: {2}", accounts.IndexOf(acct) + 1, acct.Name, acct.Location);
+                }
+                Console.WriteLine();
+
+                // Delete account
+                Console.Write("Hit ENTER to delete account {0}: ", account.Name);
+                Console.ReadLine();
+                Console.WriteLine("Deleting account {0}...", account.Name);
+                try
+                {
+                    await batchManagementClient.Account.DeleteAsync(ResourceGroupName, account.Name);
+                }
+                catch (CloudException ex)
+                {
+                    if (!(ex.Request.Method == HttpMethod.Get && ex.Message.Contains("Long running operation failed with status 'NotFound'")))
```

Comment:
💡 a quick note: `.Contains` is case sensitive, in case you want to make it case insensitive, then worth adding `OrdinalIgnoreCase` https://msdn.microsoft.com/en-us/library/dy85x1sa(v=vs.110).aspx